pkg/transforms/argoapplication.go

```diff
@@ -88,6 +88,17 @@ func ArgoApplicationResourceBuilder(a *ArgoApplication) *ArgoApplicationResource
 
 	node.Properties["applicationSet"] = applicationSet
 
+	// add its hosting applicationSet namespaced name, if the argocd app is propogated by ACM argocd pull integration controller
+	hostingApplicationSet := ""
+
+	if a.Annotations != nil {
+		if a.Annotations["apps.open-cluster-management.io/hosting-applicationset"] != "" {
+			hostingApplicationSet = a.Annotations["apps.open-cluster-management.io/hosting-applicationset"]
+		}
+	}
+
+	node.Properties["hostingApplicationSet"] = hostingApplicationSet
+
```

Comment:
Can we only add the property if not empty?

```suggestion
	// add its hosting applicationSet namespaced name, if the argocd app is propogated by ACM argocd pull integration controller

	if a.Annotations != nil {
		if a.Annotations["apps.open-cluster-management.io/hosting-applicationset"] != "" {
			node.Properties["hostingApplicationSet"] = a.Annotations["apps.open-cluster-management.io/hosting-applicationset"]
		}
	}

```

Reply:
Also could we make this property internal by adding `_` to the name?  `_hostingApplicationSet"

Reply:
fixed